### ISSUE TITLE
'gpu_metrics file does not exist!' but I found it in '/tmp/{username}/nni/script/gpu_metrics' 

### ISSUE THREAD

**Issue by pcchvol** [microsoft/nni#3299]
'gpu_metrics file does not exist!' but I found it in '/tmp/{username}/nni/script/gpu_metrics' 
and  'METRIC_OUTPUT_DIR=./ python3 -m nni_gpu_tool.gpu_metrics_collector' is
{'Timestamp': 'Wed Jan 13 14:36:07 2021', 'gpuCount': 4, 'gpuInfos': [{'index': 0, 'gpuUtil': '1', 'gpuMemUtil': '0', 'activeProcessNum': 0}, {'index': 1, 'gpuUtil': '0', 'gpuMemUtil': '0', 'activeProcessNum': 0}, {'index': 2, 'gpuUtil': '86', 'gpuMemUtil': '34', 'activeProcessNum': 1}, {'index': 3, 'gpuUtil': '91', 'gpuMemUtil': '38', 'activeProcessNum': 1}]}

**Environment**:
- NNI version:1.9
- NNI mode local
- Client OS:Ubuntu18.04
- Python version:3.7
- PyTorch version:1.6
- Is conda/virtualenv/venv used?:conda
- Is running in Docker?:False

**Log message**:
 - nnimanager.log: 
[1/13/2021, 2:30:39 PM] INFO [ 'Datastore initialization done' ]
[1/13/2021, 2:30:39 PM] INFO [ 'RestServer start' ]
[1/13/2021, 2:30:39 PM] INFO [ 'Construct local machine training service.' ]
[1/13/2021, 2:30:39 PM] INFO [ 'RestServer base port is 8888' ]
[1/13/2021, 2:30:39 PM] INFO [ 'Rest server listening on: http://0.0.0.0:8888' ]
[1/13/2021, 2:30:40 PM] INFO [ 'NNIManager setClusterMetadata, key: local_config, value: {"useActiveGpu":true,"gpuIndices":"1","maxTrialNumPerGpu":1}' ]
[1/13/2021, 2:30:40 PM] INFO [ 'Specified GPU indices: 1' ]
[1/13/2021, 2:30:40 PM] INFO [ 'NNIManager setClusterMetadata, key: trial_config, value: {"command":"python train_nni.py","codeDir":"/home/server/pyc/class/.","gpuNum":1}' ]
[1/13/2021, 2:30:40 PM] INFO [ 'required GPU number is 1' ]
[1/13/2021, 2:30:40 PM] INFO [ 'Starting experiment: snnuqajn' ]
[1/13/2021, 2:30:40 PM] INFO [ 'Change NNIManager status from: INITIALIZED to: RUNNING' ]
[1/13/2021, 2:30:40 PM] INFO [ 'Add event listeners' ]
[1/13/2021, 2:30:40 PM] INFO [ 'Run local machine training service.' ]
[1/13/2021, 2:30:40 PM] WARNING [ 'gpu_metrics file does not exist!' ]
[1/13/2021, 2:30:41 PM] INFO [ 'NNIManager received command from dispatcher: ID, ' ]
[1/13/2021, 2:30:41 PM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"img_size": 512, "lr": 0.001, "optimizer": "Adam", "rho": 0.04446677503503882, "momentum": 0.6, "weight_decay": 5e-05, "SAM": 1, "scheduler": "StepLR", "step_size": 3, "smooth_factor": 0.2, "btloss": 0, "t1": 0.95, "t2": 8, "Balance": 1, "RandomHorizontalFlip": 0, "RandomVerticalFlip": 0.25, "RandomAffine": 5, "ColorJitter": 0.1, "RandomRotation": 10, "RandomErasing": 0.5, "RandomErasingMaxScale": 0.01}, "parameter_index": 0}' ]
[1/13/2021, 2:30:45 PM] INFO [ 'submitTrialJob: form: {"sequenceId":0,"hyperParameters":{"value":"{\\"parameter_id\\": 0, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"img_size\\": 512, \\"lr\\": 0.001, \\"optimizer\\": \\"Adam\\", \\"rho\\": 0.04446677503503882, \\"momentum\\": 0.6, \\"weight_decay\\": 5e-05, \\"SAM\\": 1, \\"scheduler\\": \\"StepLR\\", \\"step_size\\": 3, \\"smooth_factor\\": 0.2, \\"btloss\\": 0, \\"t1\\": 0.95, \\"t2\\": 8, \\"Balance\\": 1, \\"RandomHorizontalFlip\\": 0, \\"RandomVerticalFlip\\": 0.25, \\"RandomAffine\\": 5, \\"ColorJitter\\": 0.1, \\"RandomRotation\\": 10, \\"RandomErasing\\": 0.5, \\"RandomErasingMaxScale\\": 0.01}, \\"parameter_index\\": 0}","index":0}}' ]
[1/13/2021, 2:30:55 PM] INFO [ 'Trial job tGTsO status changed from WAITING to FAILED' ]

**Comment by pcchvol** [comment]
my YAML:
authorName: default
experimentName: cldc
trialConcurrency: 1
maxExecDuration: 24h
maxTrialNum: 100
#choice: local, remote, pai
trainingServicePlatform: local
searchSpacePath: search_space.json
#choice: true, false
useAnnotation: false
tuner:
  #choice: TPE, Random, Anneal, Evolution, BatchTuner, MetisTuner, GPTuner
  #SMAC (SMAC should be installed through nnictl)
  builtinTunerName: TPE
  classArgs:
    #choice: maximize, minimize
    optimize_mode: maximize
trial:
  command: python train_nni.py
  codeDir: .
  gpuNum: 1
localConfig:
  useActiveGpu: true
  gpuIndices: 1
  maxTrialNumPerGpu: 1

**Comment by SparkSnail** [comment]
Hi @pcchvol , this is a normal behavior. This warning is caused because nni_gpu_metrics process and nni_manager are two different process, nni_gpu_metrics process might run slower than nni_manager process, when nni_manager start to detect gpu_metircs file, this file might has not been generated.  As soon as the file is generated, your trial job will be scheduled, and will not report this warning any more.
Your trial job failed is caused by this warning since it has been schedulered. You could check trial.log under `~/nni-experiments/{expId}/trials/{trialId}` folder to debug tiral code.

**Comment by pcchvol** [comment]
Thank you. It's not related to this warning. My code is wrong.
